Should not be able to trigger close in `PendingOpen`

scratchscratchscratchy opened the pull request:
fixes https://github.com/itchysats/itchysats/issues/517

Triggering a settlement proposal prior to being `Open` does not make sense.

scratchscratchscratchy committed:
Should not be able to trigger close in `PendingOpen`

Triggering a settlement proposal prior to being `Open` does not make sense.

klochowicz reviewed (commented):
would you mind also emitting an error from the daemon API if this is undesired behaviour?

scratchscratchscratchy committed:
Add guard for triggering settlement proposal

It only makes sense to allow proposing a settlement when in `Open` state.
Theoretically we could argue that it should be allowed in `PendingRefund` and `PendingCommit`, but I don't see an advantage in that.

Note: This only guards the taker side, no guards were added to the maker side because it remains unclear if the deamon should contain logic for this, or we always want to handle acceptance criteria for settlement from the outside.

bonomat commented on taker-frontend/src/components/History.tsx:
```diff
@@ -97,7 +97,12 @@ const CfdDetails = ({ cfd }: CfdDetailsProps) => {
+            StateKey.PENDING_CLOSE,
+        ].includes(
             cfd.state.key,
         );
```
Would it be easier to just say: 

```
const disableCloseButton = cfd.state.getGroup() === StateGroupKey.CLOSED || !StateKey.OPEN === cfd.state.key
```

bonomat reviewed (approved):
👍
